### PR TITLE
use enums + build script + precompilation

### PR DESCRIPTION
**Pull request by KristofferC**
need to fix MKL too but no license at the moment...

### CODE REVIEW

**Comment by KristofferC** [comment]
@tkelman I am trying to get this package precompileable by using a build script and an `__init__` function (last commit).

The build script basically tries to find the library and puts the path in the `deps.jl` file. Then I use `__init__` to `global const` the library and the symbols. Is that correct procedure?

Thank you.

**Comment by tkelman** [comment]
`global const` in `__init__` is pretty broken. Set all the `const`s you need in the generated `deps.jl` file and just include it.

**Comment by KristofferC** [comment]
But on precompilation that would cache the library pointers which would be different at different `dlopen`s? I thought the reason for the `__init__` was exactly to deal with refreshing the library pointers at every package load.

File: deps/build.jl [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    "libpardiso500-GNU461-X86-64",
+    "libpardiso500-GNU472-X86-64",
+    "libpardiso500-GNU481-X86-64",
+    "libpardiso"
+]
+
+@windows_only const LIBPARDISONAMES = ["libpardiso500-WIN-X86-64.dll", "libpardiso"] 
+
+const PATH_PREFIXES = [
+    joinpath(Pkg.dir(), "Pardiso", "deps"),
```
Comment (tkelman) [inline deps/build.jl:14]: don't use Pkg.dir, use `dirname(@__FILE__)` so the package can be installed elsewhere

**Pull request merged**